include/cuco/detail/traits.hpp

```diff
@@ -30,10 +29,11 @@ struct is_std_pair_like : cuda::std::false_type {
 
 template <typename T>
 struct is_std_pair_like<T,
-                        cuda::std::void_t<decltype(std::get<0>(cuda::std::declval<T>())),
-                                          decltype(std::get<1>(cuda::std::declval<T>()))>>
-  : cuda::std::
-      conditional_t<std::tuple_size<T>::value == 2, cuda::std::true_type, cuda::std::false_type> {
+                        cuda::std::void_t<decltype(cuda::std::get<0>(cuda::std::declval<T>())),
+                                          decltype(cuda::std::get<1>(cuda::std::declval<T>()))>>
```

Comment:
Needs further check. This changes the trait behavior from std-like to libcudacxx cuda-std-like.

Reply:
I thought the `cuda::std` version is a drop-in replacement for the host-side `std` implementation? Do we still need to support `std` namespace in this case?

Reply:
You're right: https://godbolt.org/z/xx53ozdPE

Does that mean we should also need a `cuco::pair` ctor specialized for `is_cuda_std_pair_like`? I'm confused.

Reply:
> Does that mean we should also need a cuco::pair ctor specialized for is_cuda_std_pair_like? 

That would be useful. We can do this in a separate PR.

Reply:
Already pushed the change when I saw your comment 🙈